### ISSUE TITLE
Devdependanies not installed on deploy

### ISSUE THREAD

**Issue by missingdink** [itsthatguy/generator-spire#19]
Had to move "Hapi" and "event-stream" into dependancies in order for circle to pass

**Comment by rylnd** [comment]
Adding to this: this was when deploying to heroku, which assumes [a production environment](https://devcenter.heroku.com/articles/nodejs-support#customizing-the-build-process)

**Comment by itsthatguy** [comment]
Yes. `dev-dependencies` should not be moved into `dependencies` if you need to run the build tools on your production server, you should configure heroku to do so.

**Comment by rylnd** [comment]
While this issue was fixed before @missingdink  even opened it, I think there's another issue along the lines of "Ensure that dev dependencies are actually dev dependencies." Granted that we need to compile on heroku, so we need those gulp dependencies, but anything that is dev-only (e.g. the `gulp watch` stuff) does not need to be installed on heroku and should be declared as a dev dependency.

**Comment by rylnd** [comment]
@itsthatguy reopening this because this issue just prevented me from deploying moments. I think the core of the issue is that generator-spire needs to have an opinion on how you're supposed to deploy.

By gitignoring the dist/ folder, we seem to be saying "build assets on the deployed machine," but by having _build-critical libraries_ listed as devDependencies, we seem to be saying "build assets locally, and then deploy." Which one is it?

**Comment by itsthatguy** [comment]
Put everything used for the build tools into dependencies
